torchgeo/datamodules/utils.py

```diff
@@ -43,6 +43,44 @@ def collate_fn_detection(batch: list[dict[str, Tensor]]) -> dict[str, Any]:
     return output
 
 
+def collate_fn_timeseries(
```

Comment:
* Move to torchgeo/_datasets_/utils.py (nearby stack_samples, etc.)
* Rename to temporal_collate, temporal_pad (check what other PyTorch padding functions are called)
* Add a `pad_length: int | None = None` parameter (if None, pad to max length of mini-batch, else pad to `pad_length`) pad_length or max_length is fine
* Have to truncate, either clearly document this or issue a warning